pkg/util/autopilot.go

```diff
@@ -55,13 +59,20 @@ func IsAutopilotManagedNamespace(o client.Object) bool {
 }
 
 // IsGKEAutopilotCluster returns if the cluster is an autopilot cluster.
-// It leverages the existence of the workload-defaulter mutating webhook configuration, which exists and only exists on the Autopilot clusters.
+// It determines an Autopilot cluster by the presence of autopilot webhooks.
 func IsGKEAutopilotCluster(c client.Client) (bool, error) {
-	err := c.Get(context.Background(), client.ObjectKey{Name: autopilotWebhook}, &admissionregistrationv1.MutatingWebhookConfiguration{})
-	if err == nil {
-		return true, nil
+	var autopilotWebhookNotFoundCount int
+	var err error
+	for _, autopilotWebhook := range autopilotWebhooks {
+		err = c.Get(context.Background(), client.ObjectKey{Name: autopilotWebhook}, &admissionregistrationv1.MutatingWebhookConfiguration{})
```

Comment:
Might be slightly more efficient to do a LIST instead of a GET. That way it only makes one API call.

Reply:
Two `Get` calls is okay. It does not worth the effort to optimize to save one API call.